### PR TITLE
add charliecloud

### PR DESCRIPTION
**Pull request by ngam**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ngam** [comment]
TODO: 
- [ ] skip ./configure
- [ ] add specs manually

**Comment by ngam** [comment]
all good for now, next:

- [ ] add proper testing (won't work with the missing namespaces?)
- [ ] manipulate build further for enhancements?
- [ ] add aarch+ppc

**Comment by ngam** [comment]
@conda-forge/staged-recipes this super neat and lightweight package is ready for your review. See my comment above about caveats re testing. Thanks!

File: recipes/charliecloud/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    # it also hard-tuns the flag ON for ch-image.
+    - hardcode_build.diff
+
+build:
+  number: 0
+  skip: true  # [not linux]
+  script:
+    - ./configure --prefix=$PREFIX
+    - make
+    - make install
```
Comment (ocefpaf) [inline recipes/charliecloud/meta.yaml:21]: Is there a make check we can run here?

Comment (ngam) [reply recipes/charliecloud/meta.yaml:22]: Not really, but I will add it anyway. I will be doing more proper testing once this is a feedstock, but for that `--privileged` flag will be needed. 

```
ubuntu@primary:~/charliecloud-0.25$ make check
Making check in lib
make[1]: Entering directory '/home/ubuntu/charliecloud-0.25/lib'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/home/ubuntu/charliecloud-0.25/lib'
Making check in bin
make[1]: Entering directory '/home/ubuntu/charliecloud-0.25/bin'
make[1]: Leaving directory '/home/ubuntu/charliecloud-0.25/bin'
Making check in doc
make[1]: Entering directory '/home/ubuntu/charliecloud-0.25/doc'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/home/ubuntu/charliecloud-0.25/doc'
Making check in examples
make[1]: Entering directory '/home/ubuntu/charliecloud-0.25/examples'
ln -fTs dirCb copy/dirCa/symlink-to-dirCb
ln -fTs fileDa copy/dirD/symlink-to-fileDa
ln -fTs dirEb copy/dirEa/symlink-to-dirEb
ln -fTs filey copy/dirG/s_dir2
ln -fTs diry copy/dirG/s_dir3
ln -fTs filey copy/dirG/s_file2
ln -fTs diry copy/dirG/s_file3
ln -fTs fileA copy/symlink-to-fileA
ln -fTs fileB copy/symlink-to-fileB-A
ln -fTs fileB copy/symlink-to-fileB-B
make[1]: Leaving directory '/home/ubuntu/charliecloud-0.25/examples'
Making check in misc
make[1]: Entering directory '/home/ubuntu/charliecloud-0.25/misc'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/home/ubuntu/charliecloud-0.25/misc'
Making check in packaging
make[1]: Entering directory '/home/ubuntu/charliecloud-0.25/packaging'
make[1]: Nothing to be done for 'check'.
make[1]: Leaving directory '/home/ubuntu/charliecloud-0.25/packaging'
Making check in test
make[1]: Entering directory '/home/ubuntu/charliecloud-0.25/test'
ln -fTs /tmp fixtures/symlink-to-tmp
make[1]: Leaving directory '/home/ubuntu/charliecloud-0.25/test'
make[1]: Entering directory '/home/ubuntu/charliecloud-0.25'
make[1]: Nothing to be done for 'check-am'.
make[1]: Leaving directory '/home/ubuntu/charliecloud-0.25'
```

**Comment by ngam** [comment]
@ocefpaf gentle nudge to merge this if you're satisfied with my edit above, thanks! Also tagging you in a related addition

**Pull request merged**